test/xpu_api/utilities/pairs/pairs.pair/assign_pair.pass.cpp

```diff
@@ -0,0 +1,108 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include <oneapi/dpl/type_traits>
+#include <oneapi/dpl/utility>
+
+#include "support/test_macros.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+struct NonAssignable
+{
+    NonAssignable&
+    operator=(NonAssignable const&) = delete;
+    NonAssignable&
+    operator=(NonAssignable&&) = delete;
+};
+struct CopyAssignable
+{
+    CopyAssignable() = default;
+    CopyAssignable(CopyAssignable const&) = default;
+    CopyAssignable&
+    operator=(CopyAssignable const&) = default;
+    CopyAssignable&
+    operator=(CopyAssignable&&) = delete;
```

Comment:
Having move assignment as deleted for copyable structure violates the standard. Move assignment is not needed here.
```suggestion
```